pkg/gpt/gpt.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2009-2017 The Go Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// gpt implements reading and writing of GUID Partition tables.
+// GPTs are dumped in JSON format and written in same.
+// One complication is that we frequently only want to
+// write a very small subset of a GPT. For example,
+// we might only want to change the GUID. As it happens
+// it is simpler (and more useful) just to read and write
+// the whole thing. In for a penny, in for a pound.
+package gpt
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"hash/crc32"
+	"io"
+	"log"
+	"reflect"
+
+	"github.com/google/uuid"
+)
+
+const (
+	BlockSize  = 512
+	HeaderOff  = 0x200
+	HeaderSize = 0x5c // They claim it can vary. Give me a break.
```

Comment:
I never understood why people like varying header sizes.

Reply:
sadly, no. We need to carefully distinguish the header type from the header size in case they do get clever and change that someday. Will they ever? I don't know. But they'e done weirder things.
The people who design this all assume assembly code.